R/zchunk_L225.hydrogen.R

```diff
@@ -1,237 +1,287 @@
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-   #
-# NOTE: This code uses repeat_and_add_vector
-# This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
+    # 1. Build tables for CSVs
+    # 1a. Supply sector information
+
+    # L225.Supplysector_h2: Supply sector information for hydrogen sectors
+    A25.sector %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[['Supplysector']], GCAM_region_names) ->
+      L225.Supplysector_h2
+
+    # 1b. Subsector information
+
+    # L225.SubsectorLogit_h2: Subsector logit exponents of hydrogen sectors
+    A25.subsector_logit %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorLogit']], GCAM_region_names) ->
+      L225.SubsectorLogit_h2
+
+    # L225.SubsectorShrwt_h2 and L225.SubsectorShrwtFllt_h2: Subsector shareweights of hydrogen sectors
+    if(any(!is.na(A25.subsector_shrwt$year))){
+      A25.subsector_shrwt[!is.na(A25.subsector_shrwt$year),] %>%
+        write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorShrwt']], GCAM_region_names) ->
+        L225.SubsectorShrwt_h2
+    }
+    if(any(!is.na(A25.subsector_shrwt$year.fillout))){
+      A25.subsector_shrwt[!is.na(A25.subsector_shrwt$year.fillout),] %>%
+        write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorShrwtFllt']], GCAM_region_names) ->
+        L225.SubsectorShrwtFllt_h2
+    }
+
+    if(exists("A25.subsector_interp")){
+      # L225.SubsectorInterp_h2 and L225.SubsectorInterpTo_h2: Subsector shareweight interpolation of hydrogen sectors
+      if(any(is.na(A25.subsector_interp$to.value))){
+        A25.subsector_interp[is.na(A25.subsector_interp$to.value),] %>%
+          write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorInterp']], GCAM_region_names)->
+          L225.SubsectorInterp_h2
+      }
+      if(any(!is.na(A25.subsector_interp$to.value))){
+        A25.subsector_interp[!is.na(A25.subsector_interp$to.value),] %>%
+          write_to_all_regions(LEVEL2_DATA_NAMES[['SubsectorInterpTo']], GCAM_region_names) ->
+          L225.SubsectorInterpTo_h2
+      }
+    }
+
+    # 1c. Technology information
+
+    # L225.StubTech_h2: Identification of stub technologies of hydrogen
+    # Note: assuming that technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A25.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[['Tech']], GCAM_region_names) -> L225.StubTech_h2
+    names(L225.StubTech_h2) <- LEVEL2_DATA_NAMES[['StubTech']]
```

Comment:
Emmm...seems not working. `select` picks out the common columns and drops out the columns I wanted to rename. I guess I could use 'rename' to directly rename the desired column since later on similar behavior happened in both re-written code and old code.